tests/loader_layer_tests.cpp

```diff
@@ -2090,6 +2091,73 @@ TEST(ExplicitLayers, WrapObjects) {
     }
 }
 
+TEST(ExplicitLayers, VkLayerPathEnvVar) {
+    FrameworkEnvironment env;
+    env.add_icd(TestICDDetails(TEST_ICD_PATH_VERSION_2_EXPORT_ICD_GPDPA));
+    env.get_test_icd().add_physical_device({});
+
+    const char* regular_layer_name_1 = "RegularLayer1";
+    env.add_explicit_layer(
+            ManifestLayer{}.add_layer(
+                ManifestLayer::LayerDescription{}.set_name(regular_layer_name_1).set_lib_path(TEST_LAYER_PATH_EXPORT_VERSION_2)),
+            "regular_layer_1.json");
+
+
+
+    { 
+        // when overriding the search path to a non-existent file, instance should fail to create
+        InstWrapper inst(env.vulkan_functions);
+        inst.create_info.add_layer(regular_layer_name_1);
+        set_env_var("VK_LAYER_PATH", "/home/phish3y/dev/Vulkan-Loader/build/install/etc/vulkan/explicit_layer.d/regular_layer_2.json");
+        inst.CheckCreate(VK_ERROR_LAYER_NOT_PRESENT);
+    }
+    { 
+        // when overriding the search path to an existing file, count should be 1
+        InstWrapper inst(env.vulkan_functions);
+        inst.create_info.add_layer(regular_layer_name_1);
+        set_env_var("VK_LAYER_PATH", "/home/phish3y/dev/Vulkan-Loader/build/install/etc/vulkan/explicit_layer.d/regular_layer_1.json");
```

Comment:
@charles-lunarg I have a set of four passing tests here that I thing covers validation around `VK_LAYER_PATH `supporting paths to both files and directories. The obvious problem I have here is the full path string on my machine. Any idea how I can build that path to `<repository>/build/install/etc/vulkan/explicit_layer.d` generically?

Reply:
The EnvVarICDOverrideSetup.TestOnlyLayerEnvVar uses the ManifestDiscoveryType::env_var to add stuff to the VK_LAYER_PATH env-var automatically. Something you may need to do is to modify the `TestLayerDetails` struct to include info on whether to use a direct path to a file or a folder, since TestEnvironment currently assumes VK_LAYER_PATH can only have folders when calling `TestEnvironment::add_explicit/implicit_layer()`